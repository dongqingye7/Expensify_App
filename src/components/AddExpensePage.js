import React from 'react';

const AddExpensePage = () => <div>why is not showing</div>;

export default AddExpensePage;