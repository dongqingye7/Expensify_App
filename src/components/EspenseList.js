import React from 'react';
import {connect} from 'react-redux'

const ExpenseList = (props)=> (
    <div>
        <h1>ExpenseList</h1>
        {props.expenses.length}
    </div>)
  
const mapStateToPtopos = (state)=>{
    return {
        expenses: state.expenses,
        filters: state.filters
    }
}
const ConnectedExpenseList = connect(mapStateToPtopos)(ExpenseList);
export default ConnectedExpenseList;