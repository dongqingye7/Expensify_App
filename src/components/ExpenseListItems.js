import React from "react";
import { connect } from "react-redux";
import {Link} from 'react-router-dom'

export const ExpenseListItem = ({ id, description, amount, createdAt }) => (
  <div>
    <Link to = {`/edit/${id}`}>
    <h3>{description}</h3>
    </Link>
    <p>
      {amount} - {createdAt}
    </p>

  </div>
);

const mapStateToProps = (state) => {
  return {
    expenses: getVisibleExpenses(state.expenses, state.filters),
  };
};
export default connect()(ExpenseListItem);
