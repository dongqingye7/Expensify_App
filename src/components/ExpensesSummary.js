import React, { Component } from "react";
import { connect } from "react-redux";
import selectExpensesTotal from "../selectors/expenses-total";
import getVisibleExpenses from "../selectors/expenses";
import numeral from "numeral";
import {Link} from 'react-router-dom'

export class ExpensesSummary extends Component {
  render() {
    const expenseWord = this.props.expenseCount === 1 ? "expense" : "expenses";
    const fomatedExpensesTotal = numeral(this.props.expensesTotal / 100).format(
      "$0,0.00"
    );
    return (
      <div className="page-header">
        <div className="content-container">
          <h2 className = "page-header__title">
            Viewing <span>{this.props.expenseCount}</span> {expenseWord} totalling{" "}
            <span>{fomatedExpensesTotal}</span>
          </h2>
          <div className = "page-header__actions">
            <Link className= "button" to='/create'>Add Expense</Link>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  const expenses = getVisibleExpenses(state.expenses, state.filters);
  return {
    expenseCount: expenses.length,
    expensesTotal: selectExpensesTotal(expenses),
  };
};

const ConnectedExpensesSummary = connect(mapStateToProps)(ExpensesSummary);
export default ConnectedExpensesSummary;
