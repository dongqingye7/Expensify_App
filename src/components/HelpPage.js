import React from 'react';

const HelpPage = () => <div>why is not showing</div>;

export default HelpPage;