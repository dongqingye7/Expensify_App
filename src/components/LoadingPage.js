import React, { Component } from "react";

class LoadingPage extends Component {
  render() {
    return (
      <div className="loader">
        <img className="loader__image" src="/images/loader.gif"></img>
      </div>
    );
  }
}

export default LoadingPage;
