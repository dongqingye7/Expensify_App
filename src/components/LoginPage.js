import React from "react";
import { connect } from "react-redux";
import { startLogin } from "../actions/auth";

export const LoginPage = ({ startLogin }) => (
  <div className="box-layout">
    <div className = "box-layout__box">
      <h1 className="box-layout__title">Expensify</h1>
      <p className="box-layout__word">Wonderful for tracking your personal expenses</p>
      <button className='button button--login' onClick={startLogin}>Login with Google</button>
    </div>
    
  </div>
);

// export class LoginPage extends React.Component {

//   onClick = () => {
//     this.props.startLogin();
//   };
//   render() {
//     return (
//       <div>
//         <button onClick={this.onClick}>Login</button>
//       </div>
//     );
//   }
// }

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin()),
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
