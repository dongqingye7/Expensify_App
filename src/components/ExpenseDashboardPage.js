import React from "react";
import ConnectedExpenseList from "./ExpenseList";
import ExpenseListFilters from './ExpenseListFilters';
import ExpensesSummary from './ExpensesSummary'
const ExpenseDashboardPage = () => (
  <div>
    <ExpensesSummary />
    <ExpenseListFilters />
    <ConnectedExpenseList />
  </div>
);
// store.subscribe(() => {
//     const state = store.getState();
//     const visibleExpenses = getVisibleExpenses(state.expenses, state.filters)
//   console.log(visibleExpenses);
// });
// const expenseOne = store.dispatch(
//   addExpense({ description: "Rent", amount: 1000, createdAt: 1000 })
// );
// const expenseTwo = store.dispatch(
//   addExpense({ description: "Food", amount: 300, createdAt: 20000 })
// );
// //store.dispatch(removeExpense({ id: expenseOne.expense.id }));
// store.dispatch(editExpense(expenseTwo.expense.id, { amount: 500 }));
// //store.dispatch(setTextFilter("rent"));
// // store.dispatch(setTextFilter());

//  store.dispatch(sortByAmount());
// store.dispatch(sortByDate());

// store.dispatch(setStartDate(125));
// store.dispatch(setStartDate());
// store.dispatch(setEndDate(125));
// store.dispatch(setEndDate());

const demoSTate = {
  expenses: [
    {
      id: "sss",
      description: "rent",
      note: "payment",
      amount: "1000",
      createdAt: 0,
    },
  ],
  filters: {
    text: "rental",
    sortBy: "amount",
    startDate: undefined,
    endDate: undefined,
  },
};

export default ExpenseDashboardPage;
