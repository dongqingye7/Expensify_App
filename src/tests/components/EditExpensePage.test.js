import { EditExpensePage } from "../../components/EditExpensePage";
import { shallow } from "enzyme";
import React from "react";
import expenses from "../fixtures/expenses";

let startEditExpense, startRemoveExpense, history, wrapper;

beforeEach(() => {
  startEditExpense = jest.fn();
  startRemoveExpense = jest.fn();
  history = { push: jest.fn() };
  wrapper = shallow(
    <EditExpensePage
      startEditExpense={startEditExpense}
      startRemoveExpense={startRemoveExpense}
      history={history}
      expense={expenses[0]}
    />
  );
});

test("should render EditExpensePage correctly", () => {
  expect(wrapper).toMatchSnapshot();
});

test("should handle editExpense", () => {
  const updates = {
    ...expenses[0],
    amount: 1997,
  };
  wrapper.find("ExpenseForm").prop("onSubmit")(updates);
  expect(history.push).toHaveBeenLastCalledWith("/");
  expect(startEditExpense).toHaveBeenLastCalledWith(updates.id, updates);
});

test("should handle removeExpense", () => {
  const id = expenses[0].id;
  wrapper.find("button").prop("onClick")();
  expect(history.push).toHaveBeenLastCalledWith("/");
  expect(startRemoveExpense).toHaveBeenLastCalledWith({ id });
});
