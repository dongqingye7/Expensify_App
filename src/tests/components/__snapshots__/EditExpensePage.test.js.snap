// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EditExpensePage correctly 1`] = `
<div>
  <div
    className="page-header"
  >
    <div
      className="content-container"
    >
      <h2
        className="page-header__title"
      >
        Edit Expense
      </h2>
    </div>
  </div>
  <div
    className="content-container"
  >
    <ExpenseForm
      expense={
        Object {
          "amount": 35,
          "createdAt": 0,
          "description": "food",
          "id": "1",
          "note": "",
        }
      }
      onSubmit={[Function]}
    />
    <button
      className="button button--remove"
      onClick={[Function]}
    >
      Remove Expense
    </button>
  </div>
</div>
`;
